sycl/source/detail/global_handler.cpp

```diff
@@ -381,7 +381,12 @@ extern "C" __SYCL_EXPORT BOOL WINAPI DllMain(HINSTANCE hinstDLL,
                    // release.
 #endif
 
-    shutdown_win();
+    try {
+      shutdown_win();
+    } catch (std::exception &e) {
+      __SYCL_REPORT_EXCEPTION_TO_STREAM("exception in shutdown_win", e);
+      return FALSE;
```

Comment:
are we sure about returning FALSE? The MS Docs here ( https://learn.microsoft.com/en-us/windows/win32/dlls/dllmain ) say that 
`When the system calls the DllMain function with any value other than DLL_PROCESS_ATTACH, the return value is ignored.`

I'm not sure where the reference is, but I think I read elsewhere that returning false when detaching can lead to the OS not reclaiming memory.  But I can't find the ref.

Reply:
Yes, returning false is correct here according to https://learn.microsoft.com/en-us/windows/win32/dlls/dllmain#return-value
Basically, the false return value means it failed to detach properly.